drivers/soundwire/stream.c

```diff
@@ -635,12 +635,17 @@ static int sdw_notify_config(struct sdw_master_runtime *m_rt)
  *
  * @bus: SDW bus instance
  */
-static int sdw_program_params(struct sdw_bus *bus)
+static int sdw_program_params(struct sdw_bus *bus, bool prepare)
 {
 	struct sdw_master_runtime *m_rt;
 	int ret = 0;
 
 	list_for_each_entry(m_rt, &bus->m_rt_list, bus_node) {
+
+		if (!prepare &&
```

Comment:
nitpick: this could do with a comment explaining the state check.

Reply:
yes, we need to explain this better, will fix.